address-space-controller/src/main/java/io/enmasse/controller/TemplateInfraResourceFactory.java

```diff
@@ -43,11 +50,20 @@ public TemplateInfraResourceFactory(Kubernetes kubernetes, SchemaProvider schema
         AddressSpacePlan plan = addressSpaceResolver.getPlan(addressSpaceType, addressSpace);
         ResourceDefinition resourceDefinition = addressSpaceResolver.getResourceDefinition(plan);
 
+
         if (resourceDefinition != null && resourceDefinition.getTemplateName().isPresent()) {
-            Map<String, String> parameters = new HashMap<>();
             AuthenticationService authService = addressSpace.getAuthenticationService();
             AuthenticationServiceResolver authResolver = authResolverFactory.getResolver(authService.getType());
 
+            String kcIdpHint = null;
+            if (addressSpace.getAuthenticationServiceKcIdpHint() != null) {
+                kcIdpHint = addressSpace.getAuthenticationServiceKcIdpHint();
```

Comment:
Using annotations, this would instead be:

if (addressSpace.getAnnotation(AnnotationKeys.KC_IDP_HINT) != null) {
   //
}

Then the additions in AddressSpace.java, AddressSpaceDeserializer etc. can be removed.

Reply:
Thanks for the good advice, that sounds much less intrusive.   I actually pushed this up to be my branch so soon.   It is not in a working state yet.